### DISCUSSION TITLE
[feature request] magick display refresh using SIGUSR1

### DISCUSSION THREAD

**Discussion by andykais** [ImageMagick/ImageMagick#3048] [Development]
I am creating an application that edits videos. I want to create an image preview that refreshes whenever the preview needs to be refreshed. My plan is to use the `display` command. It would be great if I could tell the display command when to refresh an image using `SIGUSR1` or `SIGUSR2`. The best solution that exists right now is `display -update 1.0 <my_img.jpg>`. The updates cannot be shortened to more than 1 second intervals though ([reported here](https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=27939)), which is not great UX.

The alternative feature request would be if the `display` command could automatically detect file changes and refresh the display accordingly, but that is likely a larger lift. This is how [Eye of Gnome](https://wiki.gnome.org/Apps/EyeOfGnome) functions.

**Comment by urban-warrior** [comment]
ImageMagick can automatically detect file changes and refresh the display.  See `-update` @ https://imagemagick.org/script/display.php.

**Reply by andykais** [reply]
Can you clarify what exactly it does? My experience is that the display command needs to be passed an update interval (which as mentioned above cannot be smaller than 1 second). The docs on the page you linked do mention that changes are detected automatically
> display will automagically detect that the input file has been changed and update the displayed image accordingly

but again, it appears that all imagemagick is doing is looping over the image every 1 second and checking if it should update the image. If there is a way to call this command without specifying an interval (or specifying one smaller than 1 second, please let me know)

**Comment by urban-warrior** [comment]
The minimum interval for `display` to check if an image has changed and refresh is 1 second.  The `display` program does not respond to USR signals. You can set the delay in hundreds of a second if you use the `x:` coder.  The `animate` program can be delayed by a hundredth of a second between image refreshes but it requires that the image sequence already exist-- it does not response to updated images.

**Reply by andykais** [reply]
> You can set the delay in hundreds of a second if you use the x: coder

could you show me what that would look like? (A lot of imagemagick's functionality is intimidating and "imagemagick x: coder" is a very overloaded thing to search on google)

**Comment by urban-warrior** [comment]
From the command-line:

```
convert logo: -set delay 2 x:
```
Unfortunately, it looks likes it only handles seconds, not sub-seconds.

**Reply by andykais** [reply]
darn. I guess this discussion can stay open as a feature request then
